ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/TemplateConfiguration.kt

```diff
@@ -0,0 +1,67 @@
+package com.revenuecat.purchases.ui.revenuecatui.data.processed
+
+import android.net.Uri
+import androidx.compose.ui.graphics.Color
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.paywalls.PaywallData
+import com.revenuecat.purchases.ui.revenuecatui.PaywallViewMode
+
+internal data class TemplateConfiguration(
+    val template: PaywallTemplate,
+    val mode: PaywallViewMode,
+    val packageConfiguration: PackageConfiguration,
+    val configuration: PaywallData.Configuration,
+    val images: Images,
+) {
+    private val darkModeColors = ColorsFactory.create(configuration.colors, true)
+    private val lightModeColors = ColorsFactory.create(configuration.colors, false)
+
+    fun getColors(isDarkTheme: Boolean): Colors {
+        return if (isDarkTheme) darkModeColors else lightModeColors
+    }
+
+    data class PackageInfo(
+        val rcPackage: Package,
+        val localization: ProcessedLocalizedConfiguration,
+        val currentlySubscribed: Boolean,
+        val discountRelativeToMostExpensivePerMonth: Double?,
+    )
+
+    data class Images(
+        val iconUri: Uri?,
+        val backgroundUri: Uri?,
+        val headerUri: Uri?,
+    )
+
+    data class Colors(
```

Comment:
It's weird that we need a whole new type. Could we reuse `PaywallData.Configuration.ColorInformation.Colors` and instead add a `PaywallColor.color` property to `PaywallColor`?

Reply:
There are 2 main advantages of creating a new type:
- This makes the colors not nullable, since it has already processed all fallbacks, whereas `PaywallData.Configuration.ColorInformation.Colors` has a lot of nullable colors. In iOS, you did this by creating an extension that processed the fallbacks instead, but that would still keep the nullable properties available, which would be confusing I think?
- If we did it with extensions, we would need to have them in the `RevenueCatUI` module. Reason is that the `Color` class we are using is defined in the compose library which is only available in that module, not the `purchases` module.

So it's doable but I kinda prefer having it as a separate type. Lmk what you think!

Reply:
Okay sounds good 👍🏻